src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -0,0 +1,271 @@
+                            options={options}
+                            searchable={false}
+                            clearable={false}
+                            className="coexpression-select-profile"
+                        />
+                    </div>
+                    <InfoIcon style={{marginLeft:3}} tooltip={<span>{this.selectedMolecularProfile.description}</span>}/>
+                </div>
+            );
+        } else {
+            return <LoadingIndicator isLoading={true}/>;
+        }
+    }
+
+    private header() {
+        return (
+            <div>
+                {this.dataSetSelector}
+            </div>
+        );
+    }
+
+    private geneTabs() {
+        if (this.selectedMolecularProfile && this.selectedEntrezGeneId !== undefined) {
+            return (
+                <div>
+                    <MSKTabs
+                        id="coexpressionTabGeneTabs"
+                        activeTabId={this.selectedEntrezGeneId + ""}
+                        onTabClick={this.onSelectGene}
+                        className="coexpressionTabGeneTabs"
+                        unmountOnHide={true}
+                    >
+                        {this.props.genes.map((gene:Gene, i:number)=>{
+                            return (
+                                <MSKTab
+                                    className={`coExpressionTab${i}`}
+                                    key={i}
+                                    id={gene.entrezGeneId+""}
+                                    linkText={gene.hugoGeneSymbol}
+                                >
+                                </MSKTab>
+                            );
+                        })}
+                    </MSKTabs>
+                    <CoExpressionViz
+                        coExpressionCache={this.coExpressionCache}
+                        allDataRequested={this.isAllDataRequestedForCurrentTab()}
+                        setHighlightedCoExpression={this.setHighlightedCoExpressionForCurrentTab}
+                        highlightedCoExpression={this.getHighlightedCoExpressionForCurrentTab()}
+                        gene={this.props.genes.find(g=>(g.entrezGeneId === this.selectedEntrezGeneId))!}
+                        molecularProfile={this.selectedMolecularProfile!}
+                        numericGeneMolecularDataCache={this.props.numericGeneMolecularDataCache}
+                        mutationCache={this.hasMutationData ? this.props.mutationCache : undefined}
+                        onClickRequestAllData={this.requestAllData}
+                    />
+                </div>
+            );
+        } else {
+            return <LoadingIndicator isLoading={true}/>;
+        }
+    }
+
+    componentWillMount() {
+        this.setMolecularProfileReaction = autorun(()=>{
+            // set sourceMolecularProfile to default if its not already set
+            if(!this.selectedMolecularProfile && this.profiles.length) {
```

Comment:
curious how it would become undefined after mounting?

Reply:
you know Im actually not sure why I did this instead of onMobxPromise, because it should only happen once. You're right it wouldnt go back to being undefined.